tools/c7n_index/c7n_index/metrics.py

```diff
@@ -47,26 +51,30 @@
 
 CONFIG_SCHEMA = {
     'type': 'object',
-    'additionalProperties': True,
-    'required': ['indexer', 'accounts'],
+    'additionalProperties': False,
+    'required': ['indexer'],
     'properties': {
         'indexer': {
             'oneOf': [
                 {
                     'type': 'object',
-                    'required': ['host', 'port', 'idx_name'],
+                    'additionalProperties': False,
+                    'required': ['host', 'port'],
                     'properties': {
                         'type': {'enum': ['es']},
                         'host': {'type': 'string'},
                         'port': {'type': 'number'},
                         'user': {'type': 'string'},
                         'password': {'type': 'string'},
                         'idx_name': {'type': 'string'},
+                        'doc_type': {'type': 'string'},
+                        'kms_decrypt_password': {'type': 'string'},
```

Comment:
this feels a little odd in spelling it feels like an imperative command in a declarative config, and your using it below as a boolean not a string. how about just `password_type: kms` so we don't having to keeping having to add different parameters for every password store.